### PR TITLE
releases, toc: add 3.1.0 GA release notes

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add 3.1 GA release notes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: releases/3.1.0-ga.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+title: TiDB 3.1 GA Release Notes
+category: Releases
+---
+
+# TiDB 3.1 GA Release Notes
+
+发版日期：2020 年 4 月 2 日
```
Comment (ran-huang) [inline releases/3.1.0-ga.md:8]:
```suggestion
发版日期：2020 年 4 月 16 日
```

File: releases/3.1.0-ga.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
++ TiKV
+
+    - 修复 replica read 导致 TiKV panic 的问题 [#7418](https://github.com/tikv/tikv/pull/7418) [#7369](https://github.com/tikv/tikv/pull/7369)
+    - 修复 restore 产生许多空 region 的问题 [#7419](https://github.com/tikv/tikv/pull/7419)
+    - 修复重复的 resolve lock 请求可能会破坏悲观事务原子性的问题 [#7389](https://github.com/tikv/tikv/pull/7389)
+
++ TiFlash
+
+    - 修复从TiDB同步 schema 时，进行 rename table 时潜在的问题
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:78]:
```suggestion
    - 修复从 TiDB 同步 schema 时，进行 rename table 时潜在的问题
```

File: releases/3.1.0-ga.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Bug 修复
+
++ TiDB
+
+    - 修复 merge join 在某些场景下 panic 的问题 [#15920](https://github.com/pingcap/tidb/pull/15920)
+    - 修复在计算选择率时重复考虑某些表达式的问题 [#16052](https://github.com/pingcap/tidb/pull/16052)
+    - 修复极端情况下 load 统计信息可能出现的 panic 的问题 [#15710](https://github.com/pingcap/tidb/pull/15710)
+    - 修复 SQL query 中存在等价表达式在某些情况下无法识别导致报错的问题 [#16015](https://github.com/pingcap/tidb/pull/16015)
+    - 修复从一个 database 中查询另一个 database 的 view 时报错的问题    [#15867](https://github.com/pingcap/tidb/pull/15867)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:56]:
```suggestion
    - 修复从一个数据库中查询另一个数据库的 `view` 时报错的问题 [#15867](https://github.com/pingcap/tidb/pull/15867)
```

File: releases/3.1.0-ga.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复 fast analyze handle 列时 panic 的问题 [#16080](https://github.com/pingcap/tidb/pull/16080)
+    - 修复 `current_role` 输出结果字符集不正确的问题 [#16084](https://github.com/pingcap/tidb/pull/16084)
+    - 完善 MySQL 连接握手错误相关日志 [#15799](https://github.com/pingcap/tidb/pull/15799)
+    - 修复加载审计插件后端口探测活动导致 panic 的问题 [#16065](https://github.com/pingcap/tidb/pull/16065)
+    - 修复因 TypeNull 类被错误识别为变长类型，导致 left join 上的 sort 算子 panic 的问题 [#15739](https://github.com/pingcap/tidb/pull/15739)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:61]:
```suggestion
    - 修复因 `TypeNull` 类被错误识别为变长类型，导致 left join 上的 `sort` 算子 panic 的问题 [#15739](https://github.com/pingcap/tidb/pull/15739)
```

File: releases/3.1.0-ga.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复监控 session 重试错误计数不准确的问题 [#16120](https://github.com/pingcap/tidb/pull/16120)
+    - 修复在 `ALLOW_INVALID_DATES` 模式下，weekday 结果出错的问题 [#16171](https://github.com/pingcap/tidb/pull/16171)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:63]:
```suggestion
    - 修复在 `ALLOW_INVALID_DATES` 模式下，`weekday` 结果出错的问题 [#16171](https://github.com/pingcap/tidb/pull/16171)
```

File: releases/3.1.0-ga.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复在集群中存在 TiFlash 节点时，GC 可能不能正常工作的问题 [#15761](https://github.com/pingcap/tidb/pull/15761)
+    - 修复创建 hash 分区表时指定非常大的分区数量导致 TiDB OOM 的问题 [#16219](https://github.com/pingcap/tidb/pull/16219)
+    - 让 union 语句的行为和 select 语句保持相同，修复把 warnings 当 error 的问题 [#16138](https://github.com/pingcap/tidb/pull/16138)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:66]:
```suggestion
    - 让 `union` 语句的行为和 `select` 语句保持相同，修复把 warnings 当 error 的问题 [#16138](https://github.com/pingcap/tidb/pull/16138)
```

File: releases/3.1.0-ga.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复 TopN 下推到 mocktikv 中的执行错误 [#16200](https://github.com/pingcap/tidb/pull/16200)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:67]:
```suggestion
    - 修复 `TopN` 下推到 mocktikv 中的执行错误 [#16200](https://github.com/pingcap/tidb/pull/16200)
```

File: releases/3.1.0-ga.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    - 增大 `chunk.column.nullBitMap` 的初始化长度，以避免多余的 `runtime.growslice` 开销 [#16142](https://github.com/pingcap/tidb/pull/16142)
+
++ TiKV
+
+    - 修复 replica read 导致 TiKV panic 的问题 [#7418](https://github.com/tikv/tikv/pull/7418) [#7369](https://github.com/tikv/tikv/pull/7369)
+    - 修复 restore 产生许多空 region 的问题 [#7419](https://github.com/tikv/tikv/pull/7419)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:73]:
```suggestion
    - 修复 restore 产生许多空 Region 的问题 [#7419](https://github.com/tikv/tikv/pull/7419)
```

File: releases/3.1.0-ga.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复重复的 resolve lock 请求可能会破坏悲观事务原子性的问题 [#7389](https://github.com/tikv/tikv/pull/7389)
+
++ TiFlash
+
+    - 修复从TiDB同步 schema 时，进行 rename table 时潜在的问题
+    - 默认禁用粗粒度索引过滤器
+    - 修复多数据路径配置下进行 rename table 会导致数据丢失的问题
+    - 修复某些场景下 TiFlash 存储空间上报错误的问题
+    - 修复开启 region merge 情况下从 TiFlash 读取时潜在的问题
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:82]:
```suggestion
    - 修复开启 Region Merge 情况下从 TiFlash 读取时潜在的问题
```

File: releases/3.1.0-ga.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
++ Tools
+
+    - TiDB Binlog
+
+        * 修复因为 TiFlash 相关的 DDL job 导致 Drainer 同步中断的问题 [#948](https://github.com/pingcap/tidb-binlog/pull/948) [#942](https://github.com/pingcap/tidb-binlog/pull/942)
+    - BR 
+        * 修复关闭 checksum 情况下，仍然执行 checksum 的问题[#223](https://github.com/pingcap/br/pull/223)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:90]:
```suggestion
        * 修复关闭 checksum 情况下，仍然执行 checksum 的问题 [#223](https://github.com/pingcap/br/pull/223)
```

File: releases/3.1.0-ga.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+    - Backup & Restore (BR)
+
+        * 3.1 GA 版本 BR 不支持在 3.1 GA 之前版本的 TiKV 集群上进行恢复 [#233](https://github.com/pingcap/br/pull/233)
+
+## 新功能
+
++ TiDB
+
+    - 支持在 `explain format = “dot”` 中展示 coprocessor 任务的信息 [#16125](https://github.com/pingcap/tidb/pull/16125)
```
[the remark this answers is not in the document]
Comment (ran-huang) [reply releases/3.1.0-ga.md:30]: It should be English quotation. Updated.

File: TOC.md [hunk 12]
```diff
@@ -381,6 +381,7 @@
     - [4.0.0-beta.1](/releases/4.0.0-beta.1.md)
     - [4.0.0-beta](/releases/4.0.0-beta.md)
   + v3.1
+    - [3.1.0-ga](/releases/3.1.0-ga.md)
```
Comment (ran-huang) [inline TOC.md:384]: @TomShawn 3.0 GA uses upper-case `GA`. What do you suggest?

Comment (TomShawn) [reply TOC.md:384]: Yes, you are right.

**Comment by TomShawn** [comment]
@ran-huang Please update `/reference/releases/rn.md`

**Comment by ran-huang** [comment]
> @ran-huang Please update `/reference/releases/rn.md`

@TomShawn Updated. PTAL~

**Comment by TomShawn** [comment]
LGTM

File: releases/3.1.0-ga.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
++ TiKV
+
+    - 修复 replica read 导致 TiKV panic 的问题 [#7418](https://github.com/tikv/tikv/pull/7418) [#7369](https://github.com/tikv/tikv/pull/7369)
+    - 修复 restore 产生许多空 Region 的问题 [#7419](https://github.com/tikv/tikv/pull/7419)
+    - 修复重复的 resolve lock 请求可能会破坏悲观事务原子性的问题 [#7389](https://github.com/tikv/tikv/pull/7389)
+
++ TiFlash
+
+    - 修复从 TiDB 同步 schema 时，进行 rename table 时潜在的问题
+    - 默认禁用粗粒度索引过滤器
```
Comment (scsldb) [inline releases/3.1.0-ga.md:79]: Suggest moving to feature

File: releases/3.1.0-ga.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
++ TiDB
+
+    - 支持 TiDB 在启动服务时，在开启 `report-status` 配置项情况下，如果发现 HTTP 监听端口不可用，则直接退出启动 [#16291](https://github.com/pingcap/tidb/pull/16291)
+
++ Tools
+
+    - Backup & Restore (BR)
+
+        * 3.1 GA 版本 BR 不支持在 3.1 GA 之前版本的 TiKV 集群上进行恢复 [#233](https://github.com/pingcap/br/pull/233)
```
Comment (scsldb) [inline releases/3.1.0-ga.md:24]: Suggest:
BR 不支持在 3.1 GA 版本之前的集群上进行恢复

File: releases/3.1.0-ga.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    - 支持在 `explain format = "dot"` 中展示 coprocessor 任务的信息 [#16125](https://github.com/pingcap/tidb/pull/16125)
+    - 通过 `disable-error-stack` 配置项减少日志的冗余 stack 信息 [#16182](https://github.com/pingcap/tidb/pull/16182)
+
++ Placement Driver (PD)
+
+    - 优化热点 Region 调度 [#2342](https://github.com/pingcap/pd/pull/2342)
+
++ TiFlash
+
+    - 添加 DeltaTree 引擎读写负载相关 metrics 上报
```
Comment (scsldb) [inline releases/3.1.0-ga.md:39]: Suggest:
添加上报 DeltaTree 引擎读写负载相关 metrics 信息

**Comment by ran-huang** [comment]
@scsldb @TomShawn Updated. PTAL.

**Comment by scsldb** [comment]
LGTM

File: releases/3.1.0-ga.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+    - 修复多数据路径配置下进行 rename table 会导致数据丢失的问题
+    - 修复某些场景下 TiFlash 存储空间上报错误的问题
+    - 修复开启 Region Merge 情况下从 TiFlash 读取时潜在的问题
+
++ Tools
+
+    - TiDB Binlog
+
+        * 修复因为 TiFlash 相关的 DDL job 导致 Drainer 同步中断的问题 [#948](https://github.com/pingcap/tidb-binlog/pull/948) [#942](https://github.com/pingcap/tidb-binlog/pull/942)
+    - BR
```
Comment (lilin90) [inline releases/3.1.0-ga.md:89]:
```suggestion

    - Backup & Restore (BR)
```

Comment (lilin90) [reply releases/3.1.0-ga.md:89]: To avoid possible display errors on the PingCAP website.

Comment (lilin90) [inline releases/3.1.0-ga.md:89]:
```suggestion

    - BR
    
```

Comment (lilin90) [reply releases/3.1.0-ga.md:89]: To avoid possible display errors, please leave a blank line both before and after a list.

**Pull request merged**